src/Symfony/Component/EventDispatcher/EventDispatcher.php

```diff
@@ -80,7 +80,7 @@ public function getListeners($eventName = null)
      */
     public function hasListeners($eventName = null)
     {
-        return (bool) count($this->getListeners($eventName));
+        return (bool) $this->getListeners($eventName);
```

Comment:
If you plan to replace current implementation with collection (which I doubt, therefore premature optimization), at that point it should be replaced with `->isEmpty` then, not `count`. You don't need to know the count. Contract specifies you don't support this use case anyway. If you did, interface for `getListeners` would specify `array|\Countable`.